controllers/remote/cluster_cache_tracker.go

```diff
@@ -321,16 +325,17 @@ func (t *ClusterCacheTracker) newClusterAccessor(ctx context.Context, cluster cl
 		config.CAFile = inClusterConfig.CAFile
 		config.Host = inClusterConfig.Host
 
-		// Create a new client and overwrite the previously created client.
-		c, _, cache, err = t.createClient(ctx, config, cluster, indexes)
-		if err != nil {
-			return nil, errors.Wrap(err, "error creating client for self-hosted cluster")
-		}
 		log.Info(fmt.Sprintf("Creating cluster accessor for cluster %q with in-cluster service %q", cluster.String(), config.Host))
 	} else {
 		log.Info(fmt.Sprintf("Creating cluster accessor for cluster %q with the regular apiserver endpoint %q", cluster.String(), config.Host))
 	}
 
+	// Create a client and a cache for the cluster.
+	c, cache, err = t.createCachedClient(ctx, config, cluster, indexes)
```

Comment:
```suggestion
	c, cache, err := t.createCachedClient(ctx, config, cluster, indexes)
```

And remove the variable declaration above?

Reply:
Agree!